cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationKeyStoreExtractor.java

```diff
@@ -254,6 +263,29 @@ private static KeyStore retrieveExistingKeyStore(
         }
     }
 
+    @Nonnull
+    static KeyStore createNewKeyStoreFromPem( @Nonnull final String data, @Nullable final String password )
+    {
+        try {
+            final String s = new String(Base64.getDecoder().decode(data), StandardCharsets.UTF_8).trim();
+
+            final Pattern p = Pattern.compile("[-]+BEGIN CERTIFICATE[-]+.*[-]+END CERTIFICATE[-]+", Pattern.DOTALL);
+            final Matcher match = p.matcher(s);
+            final String key;
+
+            if( !match.find() || (key = (s.substring(0, match.start()) + s.substring(match.end())).trim()).isEmpty() ) {
```

Comment:
(minor) I would find it hard to interpret this error message, and the if condition is also a bit hard to read. 

How about splitting it out somewhat like this:
```suggestion
            if( !match.find() )
            {
              throw new IllegalArgumentException("PEM format cannot be parsed: No certificate entry found.");
            }
            key = (s.substring(0, match.start()) + s.substring(match.end())).trim())
            if (key.isEmpty() ) {
                  throw new IllegalArgumentException("PEM format cannot be parsed: No private key entry found.");
             }
```

(yes it is a bit longer, but also easier to read)